hack/arktos-up-scale-out-poc.sh

```diff
@@ -0,0 +1,609 @@
+  echo "Containerd is required for Arktos"
+  exit 1
+fi
+
+if [[ ! -e "${CONTAINERD_SOCK_PATH}" ]]; then
+  echo "Containerd socket file check failed. Please check containerd socket file path"
+  exit 1
+fi
+
+# Install simple cni plugin based on env var CNIPLUGIN (bridge, alktron) before cluster is up.
+# If more advanced cni like Flannel is desired, it should be installed AFTER the clsuter is up;
+# in that case, please set ARKTOS-NO-CNI_PREINSTALLED to any no-empty value
+source ${KUBE_ROOT}/hack/arktos-cni.rc
+
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/common.sh"
+
+kube::util::ensure-gnu-sed
+
+function usage {
+            echo "This script starts a local kube cluster. "
+            echo "Example 0: hack/local-up-cluster.sh -h  (this 'help' usage description)"
+            echo "Example 1: hack/local-up-cluster.sh -o _output/dockerized/bin/linux/amd64/ (run from docker output)"
+            echo "Example 2: hack/local-up-cluster.sh -O (auto-guess the bin path for your platform)"
+            echo "Example 3: hack/local-up-cluster.sh (build a local copy of the source)"
+}
+
+### Allow user to supply the source directory.
+GO_OUT=${GO_OUT:-}
+while getopts "ho:O" OPTION
+do
+    case ${OPTION} in
+        o)
+            echo "skipping build"
+            GO_OUT="${OPTARG}"
+            echo "using source ${GO_OUT}"
+            ;;
+        O)
+            GO_OUT=$(kube::common::guess_built_binary_path)
+            if [ "${GO_OUT}" == "" ]; then
+                echo "Could not guess the correct output directory to use."
+                exit 1
+            fi
+            ;;
+        h)
+            usage
+            exit
+            ;;
+        ?)
+            usage
+            exit
+            ;;
+    esac
+done
+
+if [ "x${GO_OUT}" == "x" ]; then
+    make -C "${KUBE_ROOT}" WHAT="cmd/kubectl cmd/hyperkube cmd/kube-apiserver cmd/kube-controller-manager cmd/workload-controller-manager cmd/cloud-controller-manager cmd/kubelet cmd/kube-proxy cmd/kube-scheduler"
+else
+    echo "skipped the build."
+fi
+
+# Shut down anyway if there's an error.
+set +e
+
+
+# name of the cgroup driver, i.e. cgroupfs or systemd
+if [[ ${CONTAINER_RUNTIME} == "docker" ]]; then
+  # default cgroup driver to match what is reported by docker to simplify local development
+  if [[ -z ${CGROUP_DRIVER} ]]; then
+    # match driver with docker runtime reported value (they must match)
+    CGROUP_DRIVER=$(docker info | grep "Cgroup Driver:" |  sed -e 's/^[[:space:]]*//'|cut -f3- -d' ')
+    echo "Kubelet cgroup driver defaulted to use: ${CGROUP_DRIVER}"
+  fi
+  if [[ -f /var/log/docker.log && ! -f "${LOG_DIR}/docker.log" ]]; then
+    ln -s /var/log/docker.log "${LOG_DIR}/docker.log"
+  fi
+fi
+
+
+
+# Ensure CERT_DIR is created for auto-generated crt/key and kubeconfig
+mkdir -p "${CERT_DIR}" &>/dev/null || sudo mkdir -p "${CERT_DIR}"
+CONTROLPLANE_SUDO=$(test -w "${CERT_DIR}" || echo "sudo -E")
+
+cleanup()
```

Comment:
function cleanup, healthcheck, print_color, warning_log, start_kubedns, start_nodelocaldns, start_kubedashboard, create_psp_policy, create_storage_class are same in arktos-up-scale-out-poc.sh and arktos-up.sh
Can they be refactored?

Reply:
please see answer below